nav2-usability/scenario-code/login-routing/router.dart

```diff
@@ -0,0 +1,253 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(BooksApp());
+}
+
+class BooksApp extends StatefulWidget {
+  @override
+  State<StatefulWidget> createState() => _BooksAppState();
+}
+
+class AppState extends ChangeNotifier {
+  bool _isLoggedIn = false;
+
+  bool get isLoggedIn => _isLoggedIn;
+
+  set isLoggedIn(bool value) {
+    _isLoggedIn = value;
+    notifyListeners();
+  }
+
+  bool _viewingBooks = false;
+
+  bool get viewingBooks => _viewingBooks;
+
+  set viewingBooks(bool value) {
+    _viewingBooks = value;
+    notifyListeners();
+  }
+}
+
+class _BooksAppState extends State<BooksApp> {
+  final BookRouterDelegate _routerDelegate = BookRouterDelegate();
+  final BookRouteInformationParser _routeInformationParser =
+  BookRouteInformationParser();
+
+  @override
+  void dispose() {
+    _routerDelegate.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      title: 'Books App',
+      routerDelegate: _routerDelegate,
+      routeInformationParser: _routeInformationParser,
+    );
+  }
+}
+
+class BookRouteInformationParser extends RouteInformationParser<AppRoutePath> {
+  @override
+  Future<AppRoutePath> parseRouteInformation(
+      RouteInformation routeInformation) async {
+    final uri = Uri.parse(routeInformation.location!);
+
+    // Handle '/'
+    if (uri.pathSegments.isEmpty) {
```

Comment:
We should do login authentication here. it is usually async request, we should have a mocked class something like this (I am not an expert on doing the auth flow)
```dart
class authentication {
  Future<bool> isSignedIn(dynamic myCookie); // Return true if the cookie contains previous login session.
  Future<String> userSignedIn(String userId, String password); // Returns a token represents the login session to be stored in the cookie
}
```